File: boilerplate-fullstack-main/boilerplate-fullstack-main/Api/Models/FiscalActivity.cs
namespace Api.Models;

public class FiscalActivity
{
    public int Id { get; set; }
    public int ActivityId { get; set; }
    public int CompanyId { get; set; }
    public int FiscalUserId { get; set; }
    public string? DocumentNumber { get; set; }
    public string? ProtocolNumber { get; set; }
    public string? Rc { get; set; }
    public string? CpfCnpj { get; set; }
    public int? UfespYear { get; set; }
    public int? UfespValue { get; set; }
    public int? Quantity { get; set; }
    public int? PointsTotal { get; set; }
    public decimal? Value { get; set; }
    public string? Observation { get; set; }
    public bool Validated { get; set; }
    public DateTime CompletionDate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? ValidatedAt { get; set; }

    public Activity? Activity { get; set; }
    public Company? Company { get; set; }
    public User? FiscalUser { get; set; }
}

File: boilerplate-fullstack-main/boilerplate-fullstack-main/WebApp/src/interfaces/productivity/FiscalActivity.ts
export interface FiscalActivityCreatePayload {
  activityId: number;
  companyId: number;
  fiscalUserId: number;
  completionDate: string;
  documentNumber?: string | null;
  protocolNumber?: string | null;
  rc?: string | null;
  cpfCnpj?: string | null;
  value?: number | null;
  quantity?: number | null;
  observation?: string | null;
}

export interface FiscalActivityRead {
  id: number;
  activityId: number;
  companyId: number;
  fiscalUserId: number;
  documentNumber?: string | null;
  protocolNumber?: string | null;
  rc?: string | null;
  cpfCnpj?: string | null;
  ufespYear?: number | null;
  ufespValue?: number | null;
  quantity?: number | null;
  pointsTotal?: number | null;
  value?: number | null;
  observation?: string | null;
  validated: boolean;
  completionDate: string;
  createdAt: string;
}
